src/k8s/pkg/component/component.go

```diff
@@ -44,52 +54,68 @@ type Component struct {
 	Status bool
 }
 
+// InitializeHelmClientConfig initializes a Helm Configuration, ensures the use of a fresh configuration
+func (r *HelmClientIntitializer) InitializeHelmClientConfig() (*action.Configuration, error) {
+	settings := cli.New()
+	settings.KubeConfig = "/etc/kubernetes/admin.conf"
+
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(
+		settings.RESTClientGetter(),
+		settings.Namespace(),
+		os.Getenv("HELM_DRIVER"),
+		logAdapter,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize action config: %w", err)
+	}
+	return actionConfig, nil
+}
+
 func logAdapter(format string, v ...any) {
 	logrus.Debugf(format, v...)
 }
 
 // NewManager creates a new Component manager instance.
-func NewManager(snap snap.Snap) (*helmClient, error) {
+func NewManager(snap snap.Snap, initializers ...HelmConfigInitializer) (*helmClient, error) {
+	var initializer HelmConfigInitializer
+
+	if len(initializers) > 0 {
+		initializer = initializers[0]
+	} else {
+		// If no initializer provided, use a default one
+		initializer = &HelmClientIntitializer{}
+	}
+
 	viper.SetConfigName("components")
 	viper.SetConfigType("yaml")
 	viper.AddConfigPath(snap.Path("k8s/components"))
-	err := viper.ReadInConfig()
-	if err != nil {
+
+	if err := viper.ReadInConfig(); err != nil {
 		return nil, err
 	}
+
 	config := make(map[string]componentDefinition)
-	err = viper.Unmarshal(&config)
+	if err := viper.Unmarshal(&config); err != nil {
+		return nil, err
+	}
```

Comment:
probably unrelated to this PR but: I wonder if we should just drop `viper` completely, then define the `map[string]componentDefinition` in the Go code. Any reason to always parse the same static yaml file? Could we do this?

Reply:
Are there any cases we would be updating the `component.yaml`, like a chart version update?

Reply:
Not for an installed snap, `component.yaml` is static and immutable. The change requires a new build, therefore it's the same as having it defined in the code